support/yara_patterns/tools/pe/x86/packers.yara

```diff
@@ -108,33 +109,15 @@ rule eziriz_dotnet_reactor_62_or_newer {
 		name = "Eziriz .NET Reactor"
 		version = "6.2.0.0 or newer"
 		extra = ".NET protection"
-	strings:
-		$s01 = "{11111-22222-20001-00001}" wide
-		$s02 = "{11111-22222-30001-00001}" wide
-		$s03 = "{11111-22222-40001-00001}" wide
-		$s04 = "{11111-22222-50001-00001}" wide
 	condition:
 		pe.number_of_sections == 3 and
 		pe.imports("mscoree.dll") and
-		all of them
-}
-
-// TODO: When retdec's YARAC will be of a newer version
-//rule eziriz_dotnet_reactor_62_or_newer {
-//	meta:
-//		tool = "P"
-//		name = "Eziriz .NET Reactor"
-//		version = "6.2.0.0 or newer"
-//		extra = ".NET protection"
-//	condition:
-//		pe.number_of_sections == 3 and
-//		pe.imports("mscoree.dll") and
-//		dotnet.number_of_user_strings > 8 and
-//		dotnet.user_strings[dotnet.number_of_user_strings - 8] == "{\x001\x001\x001\x001\x001\x00-\x002\x002\x002\x002\x002\x00-\x002\x000\x000\x000\x001\x00-\x000\x000\x000\x000\x001\x00}\x00\x00" and
-//		dotnet.user_strings[dotnet.number_of_user_strings - 6] == "{\x001\x001\x001\x001\x001\x00-\x002\x002\x002\x002\x002\x00-\x003\x000\x000\x000\x001\x00-\x000\x000\x000\x000\x001\x00}\x00\x00" and
-//		dotnet.user_strings[dotnet.number_of_user_strings - 4] == "{\x001\x001\x001\x001\x001\x00-\x002\x002\x002\x002\x002\x00-\x004\x000\x000\x000\x001\x00-\x000\x000\x000\x000\x001\x00}\x00\x00" and
-//		dotnet.user_strings[dotnet.number_of_user_strings - 2] == "{\x001\x001\x001\x001\x001\x00-\x002\x002\x002\x002\x002\x00-\x005\x000\x000\x000\x001\x00-\x000\x000\x000\x000\x001\x00}\x00\x00"
```

Comment:
It seems that there is something wrong with the new YARA rule as the following two tests are failing:
```
tools.fileinfo.detection.packers.eziriz.EzirizDotnetTest_002 (sample_6.2.0.0_dotnet.exe_ --js[..])  [ FAIL ]  (0.15s)
F
======================================================================
FAIL: test_fileinfo_json_output_is_correctly_parsed (tools.fileinfo.detection.packers.eziriz.EzirizDotnetTest_002)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tcagent/work/c1867d72f67fc407/retdec-regression-tests/tools/fileinfo/detection/packers/eziriz/test.py", line 38, in test_fileinfo_json_output_is_correctly_parsed
    self.assertEqual(self.fileinfo.output['tools'][0]['name'], 'Eziriz .NET Reactor')
AssertionError: 'Microsoft' != 'Eziriz .NET Reactor'
- Microsoft
+ Eziriz .NET Reactor


----------------------------------------------------------------------
Ran 1 test in 0.002s

FAILED (failures=1)

tools.fileinfo.detection.packers.eziriz.EzirizDotnetTest_002 (sample_6.2.9.2_dotnet.exe_ --js[..])  [ FAIL ]  (0.16s)
F
======================================================================
FAIL: test_fileinfo_json_output_is_correctly_parsed (tools.fileinfo.detection.packers.eziriz.EzirizDotnetTest_002)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tcagent/work/c1867d72f67fc407/retdec-regression-tests/tools/fileinfo/detection/packers/eziriz/test.py", line 38, in test_fileinfo_json_output_is_correctly_parsed
    self.assertEqual(self.fileinfo.output['tools'][0]['name'], 'Eziriz .NET Reactor')
AssertionError: 'Microsoft' != 'Eziriz .NET Reactor'
- Microsoft
+ Eziriz .NET Reactor


----------------------------------------------------------------------
Ran 1 test in 0.003s

FAILED (failures=1)

```

Reply:
It is worse. I think YARA rule is hitting, but for some reason we filter it out. I run it with vanilla `yara` and the rule was found.

Reply:
@s3rvac, @ladislav-zezula, @metthal
Marek clarified some issues here. This new rule is fundamentally different from other existing rules and `cpdetect` (and maybe other tools) cannot deal with it. Other rules have strings to match. But this one gets detected, but there are no matches - `cpdetect` throws it away, This is what would be considered _Heuristic_ rule and until now, we would hardcode it into a C++ source code. I will try to modify the toolchain to deal with it, but I don't know how hard it will be and how many issues I come accross.